Week 1 for Team Petsy complete

blip-lorist opened the pull request:
Project members: @lilagrc, @MisShellyMac @wangg131 and @lorainekv
- Green test suite for model validations and controller methods
- Viewable website structure for guest user

Weekend tasks
- Individual research on user auth and sessions. Looking at auth project, sandboxing, etc
- Create individual to-do list on how to implement user auth and sessions

On Monday
- We'll compare our to-do list notes to come up with solid user auth and sessions processes. Then we'll group code it so we can all see the process.

blip-lorist committed:
Created Rails app, added necessary gems to the gemfile. Keeping secrets secret with .gitignore

lilagrc committed:
Added models and controllers for User, Order, OrderItem, and controller for Session.

MisShellyMac committed:
Setting up Model/Controllers for Product/Review/Category

blip-lorist committed:
Added pry rails.

lilagrc committed:
Added rspec and model rspec files

lilagrc committed:
updated gemfile

wangg131 committed:
updating rspec install gems

wangg131 committed:
review model validations & tests

blip-lorist committed:
Worked on user rspec and validation for presence of username and email.

blip-lorist committed:
Added uniqueness rspec and validation for User model.

blip-lorist committed:
Added rspec validation for uniquness of password field.)

wangg131 committed:
added tests and validations to order

wangg131 committed:
order_item validations and tests

blip-lorist committed:
WIP product validation

blip-lorist committed:
WIP product validation

blip-lorist committed:
Merge branch 'mm/lk_product_validation' of github.com:lorainekv/C3Projects--bEtsy into mm/lk_product_validation
Snagging latest version of our fork

blip-lorist committed:
Completed price,name, part of Product Validation Tests

blip-lorist committed:
Finished Product Model Validation

lilagrc committed:
Updated schema to change user and order attributes

blip-lorist committed:
Updating fork with our product validation branch
Merge branch 'mm/lk_product_validation' into team-petsy/master

blip-lorist committed:
@wangg131 and I got a routes structure set up

MisShellyMac committed:
Created Seed Data and did two migrations to add/delete columns in users and categories table.

blip-lorist committed:
Created a view folder and file for reviews.

blip-lorist committed:
Started on a POST#create spec

MisShellyMac committed:
View more info about an individual product

lilagrc committed:
Added index action and test for order_item

wangg131 committed:
started on products & categories controllers specs

wangg131 committed:
started on products view pages

wangg131 committed:
started on categories & products controller specs. THIS COMMIT.

blip-lorist committed:
WIP trying to get a create spec to work.

wangg131 committed:
created welcome controller

wangg131 committed:
generated welcome controller

wangg131 committed:
generated welcome controller2

lilagrc committed:
Added images, welcome controller, and moved rspec controllers

wangg131 committed:
merged

lilagrc committed:
Cleaned up product show page

wangg131 committed:
link to all products and products index pg

wangg131 committed:
pulled ourselves up by our bootstraps

wangg131 committed:
go home link

lilagrc committed:
Added links to leave a review and add product to cart

lilagrc committed:
fixed merge conflict

wangg131 committed:
categories index & show

wangg131 committed:
category controller specs

blip-lorist committed:
Finally got the spec to work. It was confusing since I needed to pass in a product id, in addition to the review params hash.

blip-lorist committed:
Finished new and create methods.

blip-lorist committed:
Created a new form for the website. Will style it to have fancy stars later.

blip-lorist committed:
Added a new method spec for the reviews controller.

lilagrc committed:
fixed order_items controller tests

lilagrc committed:
added rspec tests for order_item create action

wangg131 committed:
category index/show views and controller

wangg131 committed:
fixed up products index pg WIP

wangg131 committed:
category controller specs wip

wangg131 committed:
Merge branch 'team-petsy/master' of github.com:lorainekv/C3Projects--bEtsy into team-petsy/master

blip-lorist committed:
Reviews load on products show page.

lilagrc committed:
added delete from cart functionality

MisShellyMac committed:
Finished view by vendor page

wangg131 committed:
fixed categories controller

wangg131 committed:
Updating fork
Merge branch 'team-petsy/master' of github.com:lorainekv/C3Projects--bEtsy into team-petsy/master

blip-lorist committed:
Updating fork with the ability to see reviews in product view
Merge branch 'loraine-reviews-in-product-show' into team-petsy/master

MisShellyMac committed:
trying to fix

MisShellyMac committed:
Passing Rspec for user

wangg131 committed:
bootstrappin

wangg131 committed:
Merging changes from master to master.
Merge branch 'team-petsy/master' of github.com:lorainekv/C3Projects--bEtsy into team-petsy/master

MisShellyMac committed:
deleted line from views

jnf commented on app/models/order.rb:
```diff
@@ -0,0 +1,7 @@
+  has_many :order_items
+  belongs_to :user
+
+  include ActiveModel::Validations
```
Because this inherits from `ActiveRecord::Base`, you likely don't need this `include`. Please correct me if I'm wrong.

jnf commented on app/views/order_items/index.html.erb:
```diff
@@ -0,0 +1,10 @@
+<%= render "shared/header" %>
```
Does every view (minus partials) render the shared header? If so, could it be put in the application layout instead?

jnf commented:
Hi! Only a couple comments. This PR shows some great progress. Keep it up and let me know if you have any questions! :+1: :tada:

MisShellyMac committed:
Added pics and updated seed info.

jnf commented:
I forgot to mention: please merge this PR once you've read/reviewed (both?) of my comments. Please take the lack of comments here as a testament to the quality of the work that's here so far. Yinz are doing great. Keep it up.

MisShellyMac committed:
Created Ability to Sign Up as a new vendor

lilagrc committed:
Added dashboard action and view

wangg131 committed:
checkout Order wip

lilagrc committed:
Added functionality for authenticated user to add a category

wangg131 committed:
order checkout

lilagrc committed:
WIP

lilagrc committed:
WIP on checkout

blip-lorist committed:
Sessions controller spec for the create method.

lilagrc committed:
added more styling

blip-lorist committed:
Removed pry line.

blip-lorist committed:
delete #destroy spec added for sessions_controller.rb

lilagrc committed:
Began functionality for merchant to add a new product

blip-lorist committed:
Green dashboard access and redirect specs.

MisShellyMac committed:
working on having an order # persist in database.

MisShellyMac committed:
hid a function for testing purposes

MisShellyMac committed:
Merge branch 'team-petsy/master' of github.com:lorainekv/C3Projects--bEtsy into team-petsy/master

MisShellyMac committed:
Fixed typo in seed data

lilagrc committed:
Added guard clause against checking out empty cart

wangg131 committed:
routes

wangg131 committed:
Merge branch 'team-petsy/master' of github.com:lorainekv/C3Projects--bEtsy into team-petsy/master

lilagrc committed:
Changed photo upload to text instead of file upload

wangg131 committed:
view orders on dashboard view

wangg131 committed:
orders index & show by user

MisShellyMac committed:
Added receipt view info of qt, price and total

blip-lorist committed:
Wrote spec for changing status to paid during transaction.

lilagrc committed:
Added some more product controller tests

lilagrc committed:
added two category rspec tests

wangg131 committed:
users order fulfillment wip

wangg131 committed:
Merge branch 'users-c-p' into team-petsy/master

lilagrc committed:
Added product form partial and bootstrap styling

blip-lorist committed:
Spec for resetting session after transaction.

blip-lorist committed:
Merge branch 'cart-testing' into team-petsy/master
Merging branch for order session testing.

MisShellyMac committed:
Added link_to capability for item name

MisShellyMac committed:
Merge branch 'team-petsy/master' of github.com:lorainekv/C3Projects--bEtsy into team-petsy/master

lilagrc committed:
New merge
Merge branch 'team-petsy/master' of github.com:lorainekv/C3Projects--bEtsy into team-petsy/master

MisShellyMac committed:
changed time format for local time zone for receipt view.

blip-lorist committed:
Reduces stock after a purchase is complete.

wangg131 committed:
working order show for users

wangg131 committed:
Merge branch 'wang-orders2' into team-petsy/master

lilagrc committed:
Minor changes to category and added status column to products

wangg131 committed:
forgotten changes to user orders and controllers

wangg131 committed:
Merge branch 'forgotten-changes' into team-petsy/master

wangg131 committed:
schema changes

blip-lorist committed:
Added a status validation for product.

blip-lorist committed:
Merge branch 'team-petsy/master' of github.com:lorainekv/C3Projects--bEtsy into team-petsy/master

lilagrc committed:
created join table for products and categories

wangg131 committed:
schema change to allow for long CC nums

wangg131 committed:
tests for orders show/index

lilagrc committed:
figured out join tables and functionality to add multiple categories

lilagrc committed:
Added additional styling

lilagrc committed:
More styling

lilagrc committed:
Changed header links

wangg131 committed:
products#new tests

wangg131 committed:
order_items#new test

wangg131 committed:
order controllers test

wangg131 committed:
products#new tests

wangg131 committed:
orders#show specs pt2

wangg131 committed:
order fulfillment & revenue

blip-lorist committed:
Added a status field to the new/edit form for product.

lilagrc committed:
Fixed merge issues

wangg131 committed:
merging branches

blip-lorist committed:
Updated gitignore.

blip-lorist committed:
Merge branch 'team-petsy/master' of github.com:lorainekv/C3Projects--bEtsy into team-petsy/master

MisShellyMac committed:
fixed double input in schema table

lilagrc committed:
updated product controller and new product form

lilagrc committed:
removed sign-up formatting to fix bug

blip-lorist committed:
Git ignore update

blip-lorist committed:
Merge branch 'team-petsy/master' of github.com:lorainekv/C3Projects--bEtsy into team-petsy/master

lilagrc committed:
Fixed order/cart bug

lilagrc committed:
Merge branch 'team-petsy/master' of github.com:lorainekv/C3Projects--bEtsy into team-petsy/master

blip-lorist committed:
Order item controller specs updated to include the new status attribute.

wangg131 committed:
user order fulfillment bug fixed

wangg131 committed:
mergin schema

wangg131 committed:
Merge branch 'wang-7/22' into team-petsy/master

blip-lorist committed:
WIP product spec

MisShellyMac committed:
Will not let product stock go negative

MisShellyMac committed:
Made error message red with bootstrap

MisShellyMac committed:
Merge branch 'negative' into team-petsy/master

lilagrc committed:
Changed dashboard action to only show current user dashboard

wangg131 committed:
schema changes

wangg131 committed:
order_item#create tracks user_id

wangg131 committed:
Merge branch 'team-petsy/master' of github.com:lorainekv/C3Projects--bEtsy into team-petsy/master

wangg131 committed:
Users can only see their own orders

wangg131 committed:
got rid of useless scope

MisShellyMac committed:
Restricted Ability to leave a review on your own product

MisShellyMac committed:
Merge branch 'team-petsy/master' of github.com:lorainekv/C3Projects--bEtsy into team-petsy/master

blip-lorist committed:
Retired items are now invisible.

blip-lorist committed:
Updated validation spec, added scope spec.

blip-lorist committed:
Repaired product controller spec for new product.

blip-lorist committed:
Repaired the product_controller_spec PATCH update spec.

lilagrc committed:
changed layout/styling on dashboard/order fulfillment

MisShellyMac committed:
fixed reviews_controller failing test

MisShellyMac committed:
Merge branch 'team-petsy/master' of github.com:lorainekv/C3Projects--bEtsy into team-petsy/master

wangg131 committed:
denying user login if session{:user_id]

wangg131 committed:
denying user signup if session[:user_id]

lilagrc committed:
Added ability to change shipping status for order_item

blip-lorist committed:
Wrote an user email validation spec.

blip-lorist committed:
Updated email validation spec.

blip-lorist committed:
Added validation for ampersand presence in email.

blip-lorist committed:
Completed spec for email format validation.

wangg131 committed:
Merge branch 'test' into team-petsy/master

lilagrc committed:
merged

MisShellyMac committed:
Fixed one failing test in orders_controller

wangg131 committed:
mergin changes order fulfillment

MisShellyMac committed:
Rspec Green Test For Orders_Controller

lilagrc committed:
Changed shipping functionality

lilagrc committed:
Updated shipped button text

wangg131 committed:
corrected DateTime for Order Placed

wangg131 committed:
marked as shipped complete

blip-lorist committed:
Minor thumbnail formatting tweaks in bootstrap.

blip-lorist committed:
Customized thumbnail class for bootcamp.

blip-lorist committed:
Square crop for all images to avoid stretching.

blip-lorist committed:
New background option.

blip-lorist committed:
Messing with background options.

lilagrc committed:
Added functionality to filter by order status

blip-lorist committed:
Created mini brand logo

blip-lorist committed:
Added cute logo to website header.

lilagrc committed:
changed the product show page

blip-lorist committed:
Created a fixed navigation bar.

blip-lorist committed:
Merge branch 'team-petsy/master' of github.com:lorainekv/C3Projects--bEtsy into team-petsy/master

MisShellyMac committed:
Added ability to have cumulative quantity

lilagrc committed:
fixed merge

lilagrc committed:
fixed typo

wangg131 committed:
added zipcode to order table

wangg131 committed:
users can only edit their own products

wangg131 committed:
added zipcode to edit order form

wangg131 committed:
Merge branch 'requirement-coverage' into team-petsy/master

blip-lorist committed:
Resized gigantor photos.

blip-lorist committed:
Merge branch 'team-petsy/master' of github.com:lorainekv/C3Projects--bEtsy into team-petsy/master

blip-lorist committed:
Added the simple cov gem for coverage reports.

wangg131 committed:
guest cant access order fulfillment

wangg131 committed:
Merge branch 'guest-access' into team-petsy/master

wangg131 committed:
deploying to heroku

blip-lorist committed:
Fixed category retired show bug.

wangg131 committed:
added a gem to trace unused routes

MisShellyMac committed:
Merge branch 'cumulative' into team-petsy/master

blip-lorist committed:
Fixed retired show bug on vendors page.

wangg131 committed:
Merge branch 'team-petsy/master' of github.com:lorainekv/C3Projects--bEtsy into team-petsy/master

lilagrc committed:
Added white space behind site content

blip-lorist committed:
Fixing a bug in product views sorted by category and vendor.
Merge branch 'retired-bug-fix' into team-petsy/master

lilagrc committed:
Changed some button styling

lilagrc committed:
Fixed quantity issue, preventing adding out of stock items to cart

lilagrc committed:
Fixed bugs

wangg131 committed:
quantity change cart wip

wangg131 committed:
wip quantity cart

wangg131 committed:
fixed update quantity cart

wangg131 committed:
working quantity cart finally

blip-lorist committed:
green specs

blip-lorist committed:
Thanks @arhx for helping fix a quantity update bug.

lilagrc committed:
Fixed a couple of cart/quantity bugs

blip-lorist committed:
Welcome controller specs.

blip-lorist committed:
Full coverage with user controller specs, all green.

blip-lorist committed:
Updated specs for increased coverage
Merge branch 'specs' into team-petsy/master

blip-lorist committed:
Green order specs.

lilagrc committed:
Updated checkout page and styled all forms

wangg131 committed:
moar seed photos

wangg131 committed:
Merge branch 'fridayyy' into team-petsy/master

blip-lorist committed:
Working specs for order_items.

blip-lorist committed:
working specs for order_items

lilagrc committed:
added form validations for checkout

lilagrc committed:
Merge branch 'team-petsy/master' of github.com:lorainekv/C3Projects--bEtsy into team-petsy/master

blip-lorist committed:
Review styling.

blip-lorist committed:
Updated homepage view.

lilagrc committed:
Made two small updates to show pages for time and CC info displays

blip-lorist committed:
Made login errors display red.
